extensions/blocks/send-a-message/whatsapp-button/edit.js

```diff
@@ -0,0 +1,260 @@
+
+		const geoFetch = await fetch( 'http://ip-api.com/json/?fields=countryCode' )
+			.then( response => {
+				if ( ! response.ok ) {
+					return false;
+				}
+
+				return response;
+			} )
+			.catch( () => {
+				return false;
+			} );
+
+		if ( geoFetch ) {
+			const geo = await geoFetch.json();
+
+			countryCodes.forEach( item => {
+				if ( item.code === geo.countryCode ) {
+					setAttributes( { countryCode: item.value } );
+				}
+			} );
+		}
+	};
+
+	useEffect( () => {
+		if ( undefined === countryCode ) {
+			getCountryCode();
+			selectBlock( clientId );
+		}
+	} );
+
+	const validatePhoneNumber = newPhoneNumber => {
+		const phoneNumberRegEx = RegExp( /^[+]?[\s./0-9]*[(]?[0-9]{1,4}[)]?[-\s./0-9]*$/, 'g' );
+
+		if ( undefined === newPhoneNumber || newPhoneNumber.length < 1 ) {
+			return false;
+		}
+
+		return phoneNumberRegEx.test( countryCode.replace( /\D/g, '' ) + newPhoneNumber );
+	};
+
+	const setBackgroundColor = color => {
+		setAttributes( { backgroundColor: color } );
+
+		if ( color === undefined || color === WHATSAPP_GREEN || color === WHATSAPP_DARK ) {
+			return setAttributes( { colorClass: 'dark' } );
+		}
+
+		setAttributes( { colorClass: 'light' } );
+	};
+
+	const renderSettingsToggle = ( isOpen, onToggle ) => {
+		const openOnArrowDown = event => {
+			if ( ! isOpen && event.keyCode === DOWN ) {
+				event.preventDefault();
+				event.stopPropagation();
+				onToggle();
+			}
+		};
+
+		return (
+			<Button
+				className="components-toolbar__control jetpack-contact-form__toggle"
+				label={ __( 'WhatsApp Button Settings' ) }
+				onClick={ onToggle }
+				onKeyDown={ openOnArrowDown }
+				icon={ renderMaterialIcon(
+					<Path d="M10.82 12.49c.02-.16.04-.32.04-.49 0-.17-.02-.33-.04-.49l1.08-.82c.1-.07.12-.21.06-.32l-1.03-1.73c-.06-.11-.2-.15-.31-.11l-1.28.5c-.27-.2-.56-.36-.87-.49l-.2-1.33c0-.12-.11-.21-.24-.21H5.98c-.13 0-.24.09-.26.21l-.2 1.32c-.31.12-.6.3-.87.49l-1.28-.5c-.12-.05-.25 0-.31.11l-1.03 1.73c-.06.12-.03.25.07.33l1.08.82c-.02.16-.03.33-.03.49 0 .17.02.33.04.49l-1.09.83c-.1.07-.12.21-.06.32l1.03 1.73c.06.11.2.15.31.11l1.28-.5c.27.2.56.36.87.49l.2 1.32c.01.12.12.21.25.21h2.06c.13 0 .24-.09.25-.21l.2-1.32c.31-.12.6-.3.87-.49l1.28.5c.12.05.25 0 .31-.11l1.03-1.73c.06-.11.04-.24-.06-.32l-1.1-.83zM7 13.75c-.99 0-1.8-.78-1.8-1.75s.81-1.75 1.8-1.75 1.8.78 1.8 1.75S8 13.75 7 13.75zM18 1.01L8 1c-1.1 0-2 .9-2 2v3h2V5h10v14H8v-1H6v3c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V3c0-1.1-.9-1.99-2-1.99z" />
+				) }
+			/>
+		);
+	};
+
+	const renderSettings = () => {
+		return (
+			<>
+				<BaseControl
+					label={ __( 'Phone Number', 'jetpack' ) }
+					help={ __(
+						'Enter the phone number you use for WhatsApp and would like to be contacted on.',
+						'jetpack'
+					) }
+					className="jetpack-whatsapp-button__phonenumber"
+				>
+					<SelectControl
+						value={ countryCode }
+						onChange={ value => setAttributes( { countryCode: value } ) }
+						options={ countryCodes }
+					/>
```

Comment:
This is a very long list, and not easy to parse.
Don't we have this kind of list with localized labels? 🤔 

Also, is there a reason why the value contains the country code (e.g. `1US`) even though we'll eventually remove it in the `save` file (`countryCode.replace( /\D+/g, '' )`?

Without the `US` part, we could sort the codes naturally (`1, 2, 10` instead of `1, 10, 2`) — even though, I'm not entirely sure it's the appropriate ordering of this kind of codes... 🤔

Reply:
I think country names would be better, but I don't believe we have anything available like that? The good part is that for most people the correct country code should auto-select. That means they won't have to parse the list.

I'm also not sure about the ordering, I think ordering 1, 2, 10 might actually make it harder to visually parse, rather than grouping by first digit?

Reply:
The value needs to contain the country code because there are duplicate country codes (+1 is both US and Canada for example).

Reply:
Ah, good point about that (also TIL US and Canada share the same code, and lots of other countries do as well!)